### PR TITLE
Update, clean up and refactor example_evo_regression script

### PR DESCRIPTION
**Pull request by mschmidt87**
This PR updates the `example_evo_regression` script. This addresses #9 partially.

@jakobj : For the "hard" example in the script, the evolution notoriously gets stuck with a constant expression. I tried a bunch of different parameters but couldn't get anywhere. Any ideas?

### CODE REVIEW

File: examples/example_evo_regression.py [hunk 1]
```diff
@@ -1,122 +1,183 @@
...
 import sympy
-from sympy.printing.dot import dotprint
+from functools import partial
 import sys
 import time
 import torch
 
+from sympy.printing.dot import dotprint
+
 sys.path.insert(0, '../')
```
Comment (jakobj) [inline examples/example_evo_regression.py]: i think we can remove this if the package is pip installable

File: examples/example_evo_regression.py [hunk 2]
```diff
@@ -1,122 +1,183 @@
...
-        # return 1. / (1. + 1. / x[:, 0] ** 2) + 1.
-        # return 1. / (1. + 1. / x[:, 0] ** 2) + 1. / (1. + 1. / x[:, 1] ** 2)  # solved, ~50k
-        # return 1. / (1. + 1. / x[:, 0] ** 4) + 1. / (1. + 1. / x[:, 1] ** 4)
-        # return 1. / (1. + 1. / x[:, 0] ** 2)
-        # return x[:, 0] ** 2 + 2 * x[:, 0] * x[:, 1] + x[:, 1] ** 2  # solved, ~3k
-
+    f_graph = graph.to_torch()
     x = torch.Tensor(n_function_evaluations, 2).uniform_(-5, 5)
-    x[torch.abs(x) < 1e-5] = 1.  # avoid infinities due to zero division
+    # x[torch.abs(x) < 1e-5] = 1.  # avoid infinities due to zero division
```
Comment (jakobj) [inline examples/example_evo_regression.py]: remove if not necessary

File: examples/example_evo_regression.py [hunk 3]
```diff
@@ -1,122 +1,183 @@
...
+    
+    Parameters
+    ----------
+    f_target : Callable
+        Target function
+
+    Returns
+    -------
+    dict
+        Dictionary containing the history of the evoluation
```
Comment (jakobj) [inline examples/example_evo_regression.py]:
```suggestion
        Dictionary containing the history of the evolution
```

File: examples/example_evo_regression.py [hunk 4]
```diff
@@ -1,122 +1,183 @@
...
-    fig = plt.figure(figsize=(width, width / scipy.constants.golden))
-    ax_fitness = fig.add_subplot(311)
-    ax_fitness.set_xlabel('Evolution step')
-    ax_fitness.set_ylabel('Fitness')
+    pop = gp.CGPPopulation(**params['population_params'],
+                           seed=params['seed'], genome_params=params['genome_params'])
 
-    ax_phenotype_distance = fig.add_subplot(312)
-    ax_genotype_distance = fig.add_subplot(313)
+    # create instance of EA
```
Comment (jakobj) [inline examples/example_evo_regression.py]:
```suggestion
    # create instance of evolutionary algorithm
```

File: examples/example_evo_regression.py [hunk 5]
```diff
@@ -1,122 +1,183 @@
...
-    # sympy.plot(sympy_expr[0])
-
-    # export computational graphs; create pdf with
-    # $ dot -Tpdf {filename} -O
-    with open('example_evo_regression-graph.dot', 'w') as f:
-        f.write(dotprint(sympy_expr[0]))
-    with open('example_evo_regression-graph_simplified.dot', 'w') as f:
-        f.write(dotprint(sympy_expr[0].simplify()))
+    obj = partial(objective, target_function=f_target)
+    # Perform evoluation
```
Comment (jakobj) [inline examples/example_evo_regression.py]:
```suggestion
    # Perform evolution
```

File: examples/example_evo_regression.py [hunk 6]
```diff
@@ -1,122 +1,183 @@
 import numpy as np
 import matplotlib.pyplot as plt
+import os
 import scipy.constants
 import sympy
-from sympy.printing.dot import dotprint
+from functools import partial
```
Comment (jakobj) [inline examples/example_evo_regression.py]: i would `import functools` to not loose the namespace on `partial`, but maybe it's common enough that it's clear where `partial` comes from

File: examples/example_evo_regression.py (hunk 1 above)
Comment (mschmidt87) [reply examples/example_evo_regression.py]: I will remove all of these calls from the scripts, incl. the test suite, once we have merged #19 . Let's do this in a separate PR.

Comment (mschmidt87) [reply examples/example_evo_regression.py]: Added a ticket for this: #26

File: examples/example_evo_regression.py (hunk 2 above)
Comment (mschmidt87) [reply examples/example_evo_regression.py]: done.

**Comment by mschmidt87** [comment]
I addressed all comments.

**Pull request merged**